code/game/objects/structures/morgue.dm

```diff
@@ -354,12 +332,58 @@ GLOBAL_LIST_EMPTY(crematoriums)
 	add_fingerprint(user)
 	update_icon(UPDATE_OVERLAYS)
 
-/obj/structure/crematorium/attackby(P as obj, mob/user as mob, params)
+/obj/structure/crematorium/obj_break(damage_flag)
+	if(!broken)
+		visible_message("<span class='warning'>[src] dims as its paneling collapses and it becomes non-functional.</span>")
+		icon_state = "crema_broke" // this will need a proper sprite when possible, as it's just a copy of crema
+		resistance_flags = INDESTRUCTIBLE // prevents it from being destroyed instead of just broken
+		name = "broken crematorium"
+		desc = "A broken human incinerator. No longer works well on barbeque nights. It requires a new igniter to be repaired."
+		repairstate = 0
+		broken = TRUE
+		cremating = FALSE
+		update_icon(UPDATE_OVERLAYS)
+		GLOB.crematoriums -= src
+
+/obj/structure/crematorium/welder_act(mob/user, obj/item/I)
+	if(user.a_intent == INTENT_HARM) // if you want to damage it with a welder you should be able to
+		return
+	if(broken)
+		if(!I.tool_use_check(user, 0))
+			return TRUE
+		if(!repairstate == 1)
+			to_chat(user, "<span class='notice'>[src] needs a new igniter before you weld the paneling closed.</span>")
+			return TRUE
+		WELDER_ATTEMPT_REPAIR_MESSAGE
+		if(!I.use_tool(src, user, 3 SECONDS, volume = I.tool_volume))
+			return TRUE
+		WELDER_REPAIR_SUCCESS_MESSAGE
+		icon_state = initial(icon_state)
+		resistance_flags = NONE
+		name = initial(name)
+		desc = initial(desc)
+		repairstate = 2
+		broken = FALSE
+		obj_integrity = max_integrity
+		GLOB.crematoriums += src
+		return TRUE
+	else
+		to_chat(user, "<span class='notice'>The crematorium does not seem to need fixing.</span>")
+		return TRUE
```

Comment:
could you move this up to become a `if(!broken)` check and early return so the code above can be unindented

Reply:
donezo